drivers/ptp/ptp_clockmatrix.c

```diff
@@ -2468,13 +2497,24 @@ static int idtcm_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, idtcm);
 
+	// add the dpll as a hw clock. this is needed to cause the phys and pcies (modems) components
+	// to be called after the dpll initialization by creating clock dependencies in the 
+	// dts file (DEV-6442)
+	err = internal_of_clk_add_hw_provider(pdev);
+	if (err)
+		return err;
+
+	printk(KERN_INFO "idtcm_probe end\n");
+
 	return 0;
 }
 
 static int idtcm_remove(struct platform_device *pdev)
 {
 	struct idtcm *idtcm = platform_get_drvdata(pdev);
 
+	ptp_clock_unregister_all(idtcm);
```

Comment:
not sure this is correct. Now we are calling it twice. We need to unregister now the clk that we registered.

Reply:
using clk_hw_unregister_fixed_rate

Reply:
fixed